analyzers/src/SonarAnalyzer.Common/Rules/BooleanLiteralUnnecessaryBase.cs

```diff
@@ -205,9 +207,10 @@ private Location CalculateExtendedLocation(SyntaxNode parent, bool isLeftSide)
             return isLeftSide ? parent.CreateLocation(token) : token.CreateLocation(parent);
         }
 
-        private static bool CheckForNullability(SyntaxNode left, SyntaxNode right, SemanticModel model) =>
-            right is null // Avoids DeclarationPattern or RecursivePattern
-            || model.GetTypeInfo(left).Type.IsNullableBoolean()
-            || model.GetTypeInfo(right).Type.IsNullableBoolean();
+        private static bool TypeShouldBeIgnored(SyntaxNode node, SemanticModel model)
+        {
+            var type = model.GetTypeInfo(node).Type;
+            return type.IsNullableBoolean() || type.Is(KnownType.System_Object);
```

Comment:
Misses 
* [interfaces implemented by boolean](https://learn.microsoft.com/en-us/dotnet/api/system.boolean?view=net-8.0) (e.g. `IComparable`),
* struct constrained and 
* unconstrained generic parameter.

see [sharplab](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQGED2AbbEDGMAlpgHYAyRMECU2yANACYgDUAPgAIBMAjALAAoTgGYABDzHoxAbyFiFE8ZwAsYgLIAKAJJYAtgAcotYHjH5Mh41FMQAlGPmK5gxW7FEwYzRasmzRMhiMAhw9k7ushHuAL7RcYLRohJqWpjAAFYEMGKYDtEukQqe3pgeQSFhdtFuhUUJbglJyqkAPAAqAHya7cH5rs41iiWaOYHBoeEDkXWRDYpN00opGrwd3b0wDgDuABY0EGLtIGLIlYQFQ8Veo+UTVVdRS43xQs0r6tzrPX1iewdHE66SxGfwQS7PYY3MYVSbVSEKWaxV6CGJAA)